app/models/workflow.rb

```diff
@@ -20,21 +20,33 @@ def self.build_given_a_hierarchy(definition)
 
     workflow.first_step = definition[:first_step]
 
-    set_up_the_method(workflow.first_step) { |e| Workflow.build_event_handler_for(workflow.first_step).fire e }
+    set_up_the_method(workflow.first_step) { |e| Workflow.build_event_handler_for(workflow.first_step, e).fire e }
 
     workflow
   end
 
   def self.set_up_the_method(step, &block)
-    step[:method] = block || lambda { |e| Workflow.build_event_handler_for(step).receive e }
+    step[:method] = block || lambda { |e| Workflow.build_event_handler_for(step, e).receive e }
     step[:config] = {} if step[:config].nil?
     return if step[:next_steps].nil?
     step[:next_steps].each { |x| set_up_the_method(x) }
   end
 
-  def self.build_event_handler_for(step)
+  def self.build_event_handler_for(step, event)
     event_handler = step[:type].constantize.new
     event_handler.config = step[:config] if event_handler.respond_to?(:config)
+
+    event_handler.config.each do |config|
+      key = config[0]
+      value = config[1]
+      if value.is_a?(String)
+        template = Liquid::Template.parse(value)
+	event_handler.config[key] = template.render(event.respond_to?(:data) ? event.data : event)
```

Comment:
This is another thing that will be affected by the spark change. Let's get `spark` in so we can keep going.